### DISCUSSION TITLE
I'm having issues with where condition can anyone help ?

### DISCUSSION THREAD

**Discussion by piyushpathar** [jenssegers/laravel-mongodb#2698] [Q&A]
I'm having issues with the Eloquent query for my Subscription model. The following code is not returning the expected results:
$id = $request->id;
$companyId = User::find($id)->company->id;
$subscription = Subscription::where('companyId', $companyId)->get();

**Comment by GromNaN** [comment]
Do you have the same type for `$companyId` and the `companyId` field in the document stored in MongoDB?

Perhaps you should cast the id into string.
```php
$subscription = Subscription::where('companyId', (string) $companyId)->get();
```
